src/video/sdl2_v.cpp

```diff
@@ -367,13 +367,18 @@ static uint ConvertSdlKeycodeIntoMy(SDL_Keycode kc)
 bool VideoDriver_SDL_Base::PollEvent()
 {
 	SDL_Event ev;
+	// workaround for old SDL libraries
+	bool is_SDL_2_0 = this->sdl_lib_major == 2 && this->sdl_lib_minor < 24;
 
 	if (!SDL_PollEvent(&ev)) return false;
 
 	switch (ev.type) {
 		case SDL_MOUSEMOTION:
-			if (_cursor.UpdateCursorPosition(ev.motion.x, ev.motion.y, true)) {
+			if (_cursor.UpdateCursorPosition(ev.motion.x, ev.motion.y, is_SDL_2_0)) {
+				SDL_SetRelativeMouseMode(is_SDL_2_0 ? SDL_FALSE : SDL_TRUE);
 				SDL_WarpMouseInWindow(this->sdl_window, _cursor.pos.x, _cursor.pos.y);
+			} else {
+				SDL_SetRelativeMouseMode(SDL_FALSE);
```

Comment:
Since the SDL version is unlikely to change from one cursor update to the next, I wonder whether a one-time config at build time would be possible instead of a runtime check?